lib/BackendClient.js

```diff
@@ -477,6 +563,23 @@ var BackendClient = (function(){
         });
 
     };
+    
+    BackendClient.getSingleEntryServerNode = function(entryId, entryServerType) {
+        let filter = new kalturaVO.KalturaEntryServerNodeFilter();
+        filter.serverTypeEqual = entryServerType;
+        return getLiveEntryServerNodes(entryId, filter)
+            .then((liveEntryServerNodes)=> {
+                if (liveEntryServerNodes.length > 0) {
+                    logger.info(`[${entryId}], received entry server node, id:[${liveEntryServerNodes[0].id}]`);
+                    return Q.resolve(liveEntryServerNodes[0]);
```

Comment:
why are you taking the first one? your live controller can acaully be using the second one..

Reply:
If we already filtered based on server type (LIVE_PRIMARY/LIVE_BACKUP) how can it be a 2nd one, however to make sure added also isLive = true and StatusIn 1,2